uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PopupViewController.cpp

```diff
@@ -62,6 +73,77 @@ PopupManager* PopupViewController::popupManager() const
   return m_popupManager;
 }
 
+/*!
+  \brief Returns the \c Popup that populates this controller with data, this takes precedence over PopupManager.
+ */
+Popup* PopupViewController::popup() const
+{
+  return m_popup;
+}
+
+/*!
+    \brief Returns the known list of available PopupElementViewControllers.
+    Internally, this is a \c GenericListModel with an \c elementType of
+    \c TextPopupElementViewController.
+ */
+GenericListModel* PopupViewController::popupElements() const
+{
+  return m_popupElementsModel;
+}
+
+/*!
+  \brief Sets the \c Popup. Setting this will trigger evaluate expressions on the popup and notify the
+  popup and title have changed.
+  \list
+  \li \a popup To deliver data from.
+  \endlist
+ */
+void PopupViewController::setPopup(Popup* popup)
+{
+  if (m_popup == popup)
+      return;
+
+  if (m_popup)
+  {
+      disconnect(m_popup.data(), nullptr, this, nullptr);
+      m_popupElementsModel->removeRows(0, m_popupElementsModel->rowCount());
+  }
+
+  m_popup = popup;
+
+  if (m_popupManager)
+      qWarning() << "Both Popup and PopupManager have been set. The PopupView will default to PopupView which supports PopupElements.";
```

Comment:
IDE formatter was fighting me on this as well because I think toolkit code styling is different from ours. I'll format and stick with toolkit styling.